app/javascript/src/warn-dsit-organisation-change.js

```diff
@@ -0,0 +1,40 @@
+/*
+  Show a standard JavaScript `confirm` modal with a warning message if a new
+  user is being created whose email address is not in the DSIT whitelist and
+  their organisation is being set to DSIT, or when an existing user is being
+  edited whose email address is not in the DIST whitelist and their
+  organisation is being set to DSIT from a different organisation.
+*/
+document.addEventListener("DOMContentLoaded", function() {
+  const VALID_DSIT_DOMAINS = ["dsit.gov.uk", "odamanagement.org"];
+  const DSIT_ORGANISATION_INDEX = 0;
+  const WARNING = "Are you sure you wish to add this user's primary organisation as DSIT?";
+
+  const userForm = document.querySelector("form.user-form");
```

Comment:
The `class` added for this purpose doesn't have a CSS counterpart. To use the subjunctive, *were* we to move things into translations via a `data: { ... }` hash, perhaps we could also use `form[data=warn-on-non-dsit]` here and remove the class from `user/_form.html.haml` to make it clear that there aren't any visual CSS classes to look for and this is pure behaviour?